libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -46,23 +41,28 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
     }
 
     if (auto status = checkDagBlockValidation(block); !status.first) {
-      // Ignore new block packets when pbft syncing
+      // Ignore new block packets when pbft syncing or dag syncing
       if (syncing_state_->is_pbft_syncing()) {
         LOG(log_dg_) << "Ignore new dag block " << hash.abridged() << ", pbft syncing is on";
         return;
       }
 
-      LOG(log_er_) << "Received NewBlock " << hash << " has missing pivot or/and tips " << status.second;
-      status.second.insert(hash);
-      requestDagBlocks(packet_data.from_node_id_, status.second, DagSyncRequestType::MissingHashes);
-      return;
+      if (peer->peer_dag_syncing) {
+        LOG(log_dg_) << "Ignore new dag block " << hash.abridged() << ", dag syncing is on";
+      } else {
+        if (peer->peer_dag_synced) {
+          LOG(log_er_) << "DagBlock" << block.getHash() << " has missing pivot or/and tips " << status.second
+                       << " . Peer " << packet_data.from_node_id_.abridged() << " will be disconnected.";
+          disconnect(peer->getId(), dev::p2p::UserReason);
```

Comment:
I think here is dangerous. DAG is growing constantly, that's very hard to determined peer DAG has synced or not

Reply:
This flag indicates that we at least once did pull all the non-finalized dag blocks from the node sending us this message. If this has already happened and if there is a missing tip/pivot node is disconnected since this should not be possible with honest node